packages/front-end/components/Experiment/TabbedPage/ExperimentStatusIndicator.tsx

```diff
@@ -1,36 +1,116 @@
-import { ExperimentStatus } from "back-end/types/experiment";
+import { ExperimentInterfaceStringDates } from "back-end/types/experiment";
+import Badge from "@/components/Radix/Badge";
 
-export interface Props {
-  status: ExperimentStatus;
-  subStatus?: string;
-}
+// Examples:
+// "full" - "Running: ~5 days left"
+// "status-only" - "Running"
+// "detail-only" - "~5 days left"
+type LabelFormat = "full" | "status-only" | "detail-only";
 
-function getColor(status: ExperimentStatus) {
-  switch (status) {
-    case "draft":
-      return "warning";
-    case "running":
-      return "info";
-    case "stopped":
-      return "secondary";
-  }
-}
+type ExperimentData = Pick<
+  ExperimentInterfaceStringDates,
+  "status" | "archived" | "results"
+>;
+
+type Props = {
+  experimentData: ExperimentData;
+  labelFormat?: LabelFormat;
+  // If true we will show the underlying status of the experiment
+  // even if the experiment is archived.
+  skipArchived?: boolean;
+};
 
 export default function ExperimentStatusIndicator({
-  status,
-  subStatus,
+  experimentData,
+  labelFormat = "full",
+  skipArchived = false,
 }: Props) {
-  const color = getColor(status);
-  return (
-    <div className="d-flex align-items-center nowrap">
-      <div
-        className={`bg-${color} rounded-circle`}
-        style={{ width: 10, height: 10 }}
-      />
-      <div className={`text-${color} ml-2`}>
-        {status}
-        {subStatus ? <> ({subStatus})</> : null}
-      </div>
-    </div>
+  const [color, variant, status, detailedStatus] = getBadgeProps(
+    experimentData,
+    skipArchived
   );
+
+  const label = getFormattedLabel(labelFormat, status, detailedStatus);
+
+  return <Badge color={color} variant={variant} radius="full" label={label} />;
+}
+
+function getBadgeProps(
+  experimentData: ExperimentData,
+  skipArchived: boolean
+): [
+  React.ComponentProps<typeof Badge>["color"],
+  React.ComponentProps<typeof Badge>["variant"],
+  string, // formattedStatus
+  string? // detailedStatus if applicable
+] {
+  if (!skipArchived && experimentData.archived) {
+    return ["gold", "soft", "Archived"];
+  }
+
+  if (experimentData.status === "draft") {
+    return ["indigo", "soft", "Draft"];
+  }
+
+  if (experimentData.status == "running") {
+    return ["indigo", "solid", "Running"];
+
+    // TODO: Add detail statuses
+    // return ["indigo", "solid", "Running", "~5 days left"];
+    // return ["amber", "solid", "Running", "Unhealthy"];
+    // return ["amber", "soft", "Running", "Rollback now"];
+    // return ["amber", "soft", "Running", "Ship now"];
+    // return ["amber", "soft", "Running", "Discuss results"];
```

Comment:
Are these detailed statuses available now or do they rely on the decision making framework?

Reply:
Yeah they rely on the decision framework -- the first one we plan on shipping is Unhealthy.

So while we don't have it at the moment, wanted to keep this list in code so we remember about it -- but also happy to remove it and reference Figma instead.

Reply:
Nah - happy to leave them in for reference, just wanted to confirm my understanding.